config_test.go

```diff
@@ -1,57 +1,165 @@
 package main
 
 import (
-	"path"
+	"bytes"
+	"log"
 	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
 )
 
 func Test_Load(t *testing.T) {
-	tt := []struct {
-		Path string
-		Err  error
-	}{
-		{
-			"valid_config.json",
-			nil,
-		}, {
-			"valid_config_extras.json",
-			nil,
-		}, {
-			"invalid_config_author.json",
-			ErrAuthorRequired,
-		}, {
-			"invalid_config_email.json",
-			ErrEmailRequired,
-		},
-	}
-
-	for _, tc := range tt {
-		path := path.Join("testdata", tc.Path)
-		_, err := LoadConfigFile(path)
-		if err != tc.Err {
-			t.Fatalf("Load(%v) returned an error (%v)", path, err)
+	Convey("When loading a valid config file", t, func() {
+		fs = mockFS{
```

Comment:
mockFS currently lives in `messages_test.go`. It exists for the same reason as the clock interface. It allows us to mock out our filesystem calls in tests but use the real implementation when running the application. It might be overkill for how simple our application is but since we want it to be open to contribute to, more tests is better in my opinion.